File: Python/Problems/Problems/problem2864.py
import heapq
import math
import sys

from utils import print1DArray, print2DArray
from typing import List, Set
from collections import Counter

from .problem import Problem


class Problem2864(Problem):
    def test(self) -> bool:
        s = "0101"

        expected = "1001"

        result = self.maximumOddBinaryNumber(s)

        print(result)

        return expected == result

    def maximumOddBinaryNumber(self, s: str) -> str:
        zeros = 0
        ones = -1
        for c in s:
            if c == '0':
                zeros += 1
            else:
                ones += 1

        return '1' * ones + '0' * zeros + '1'

File: C++/Problems/Problems/problem_2864.cpp
#include "problem_2864.h"

bool problem_2864::test() {
    std::string s = "0101";

    std::string expected = "1001";

    auto result = maximumOddBinaryNumber(s);

    std::cout << result << std::endl;

    return expected == result;
}

std::string problem_2864::maximumOddBinaryNumber(std::string s) {
    int zeros = 0;
    int ones = 0;
    for (auto c : s)
    {
        if (c == '0')
            zeros++;
        else
            ones++;
    }

    return std::string(ones - 1, '1') + std::string(zeros, '0') + "1";
}
